<launch>
      <arg name="model" default="$(find plane_simple)/urdf/plane.urdf.xacro"/>
  <arg name="gui" default="true" />


  <!-- these are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>

  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
  <arg name="world_name" value="$(find plane_simple)/worlds/myworld.world"/>
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>


  <param name="robot_description" command="$(find xacro)/xacro '$(arg model)'"/>
  <param name="use_gui" value="$(arg gui)"/>

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model"
        args="-z 1.0 -Y 2.1 -unpause -urdf -model plane -param robot_description" respawn="false" output="screen" /> 
      
    
    <rosparam command="load" file="$(find plane_simple)/param/HID.yaml" />
    <rosparam command="load" file="$(find plane_simple)/param/aerodynamics.yaml" />
    <node pkg="joy" type="joy_node" name="joystick" output="screen" >
      <param name="dev" value="/dev/input/js0" />
      <param name="deadzone" value="0.01" />
    </node>
    <node pkg="plane_simple" name="joy2chan" type="joy2chan" output="screen" />
    <node pkg="plane_simple" name="chan2basics" type="chan2basics" output="screen"/>
    <node pkg="plane_simple" name="basics2forces" type="basics2forces" output="screen" />
    
    
</launch>
    
    